src/SafeAreaContext.tsx

```diff
@@ -162,3 +202,12 @@ export const SafeAreaConsumer = SafeAreaInsetsContext.Consumer;
  * @deprecated
  */
 export const SafeAreaContext = SafeAreaInsetsContext;
+
+function useLatest<T>(value: T) {
+  const ref = React.useRef(value);
+  ref.current = value;
```

Comment:
This should be in a `useLayoutEffect`

Reply:
Oh cool I didn't know that would be beneficial here. When writing the subscription API, it changed how I did things here, which removed the need for this hook - https://github.com/th3rdwave/react-native-safe-area-context/pull/429